src/header.iced

```diff
@@ -13,60 +13,77 @@ current_minor = 0
 crypto_auth_KEYBYTES = 32
 crypto_secretbox_KEYBYTES = 32
 
-compute_mac_key = (encryptor, header_hash, pubkey) ->
-  zero_bytes = Buffer.alloc(crypto_auth_KEYBYTES)
+compute_mac_key = ({encryptor, header_hash, pubkey}, cb) ->
+  zero_bytes = new Buffer(crypto_auth_KEYBYTES)
+  zero_bytes.fill(0)
   mac_box = encryptor.encrypt({plaintext : zero_bytes, nonce : nonce.nonceForMACKeyBox(header_hash), pubkey})
   # take last crypto_auth_BYTES bytes of MAC box
-  return mac_box.slice(-crypto_auth_KEYBYTES)
+  return cb(null, mac_box.slice(-crypto_auth_KEYBYTES))
 
-exports.generate_encryption_header_packet = (encryptor, recipients) ->
+exports.generate_encryption_header_packet = ({encryptor, recipients, anonymized_recipients}, cb) ->
```

Comment:
`compute_mac_key` could error if `encryptor.encrypt` fails, which we were dropping on the ground. now it's `try/catch`ed, so `generate_encryption_header_packet` could call back with errors.